components/feature/media/src/main/java/mozilla/components/feature/media/notification/MediaNotificationFeature.kt

```diff
@@ -0,0 +1,67 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.media.notification
+
+import android.content.Context
+import android.content.Intent
+import mozilla.components.feature.media.service.MediaService
+import mozilla.components.feature.media.state.MediaState
+import mozilla.components.feature.media.state.MediaStateMachine
+import java.lang.ref.WeakReference
+
+/**
+ * Feature for displaying an ongoing notification (keeping the app process alive) while web content is playing media.
+ *
+ * This feature should get initialized globally once on app start.
+ */
+class MediaNotificationFeature(
+    private val context: Context,
+    private val stateMachine: MediaStateMachine
+) {
+    private var serviceRunning = false
+
+    /**
+     * Enables the feature.
+     */
+    fun enable() {
+        stateMachine.register(MediaObserver(this))
+    }
+
+    internal fun start(state: MediaState) {
```

Comment:
Could you rename `start` and `stop`? I assumed this was a lifecycle aware feature when I was first reading through the code.